src/requests/AuthApi.ts

```diff
@@ -0,0 +1,116 @@
+// import { VRCWrapper } from "../AuthentificationHandler";
+// import { BaseVRChatWrapper } from "../BaseVRChatWrapper";
+import { VRCWrapper } from "../VRCWrapper";
+import { CookiesReadError, CookiesUser404, CookiesWriteError, CurrentUserObjectParseError, TOTPRequired, UserNotAuthenticated } from "../errors";
+import { ApiPaths } from "../types/ApiPaths";
+import { CurrentUser, CurrentUserFromJSON, isCurrentUserJSON } from "../types/CurrentUser";
+import { BaseApi } from "./BaseApi";
+
+
+
+
+export class AuthApi extends BaseApi {
+    baseClass: VRCWrapper;
+
+    constructor(baseClass: VRCWrapper) {
+        super(baseClass);
+        this.baseClass = baseClass;
+    }
+
+    async getCurrentUser(): Promise<CurrentUser> {
+
+        if (!this.baseClass.isAuthentificated) {
+            throw new UserNotAuthenticated();
+        }
+
+        let result: CurrentUser;
+        const path = ApiPaths.auth.getCurrentUserInfo;
+        const url: URL = new URL(ApiPaths.api.base.path + path.path);
+
+        const headers = {
+            'Authorization': `Basic ${this.baseClass.getBase64Credentials()}`,
+            "User-Agent": process.env.USER_AGENT || "ExampleApp/1.0.0 Email@example.com",
+            "Cookie": `${this.baseClass.instanceCookie.getAuthCookie()}`,
+        };
+
+        const options: RequestInit = {
+            method: path.method,
+            headers: headers
+        };
+
+        try {
+
+            const response: VRCAPIResponse = await fetch(url, { ...options });
```

Comment:
You don't need to do the spread operator here, actually is even bad to do that, just pass the actual argument, doing the spread of the object you're creating a new one and then passing it, which will be just GC garbage.